File: Source/GUI/Android/app/src/main/java/net/mediaarea/mediainfo/Report.kt
/*  Copyright (c) MediaArea.net SARL. All Rights Reserved.
 *
 *  Use of this source code is governed by a BSD-style license that can
 *  be found in the License.html file in the root of the source tree.
 */

package net.mediaarea.mediainfo

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.ColumnInfo

@Entity(tableName = "reports")
data class Report(@PrimaryKey(autoGenerate = true)
                   @ColumnInfo(name = "id")
                   val id: Int,
                   @ColumnInfo(name = "filename")
                   val filename: String,
                   @ColumnInfo(name = "report", typeAffinity = ColumnInfo.BLOB)
                   val report: ByteArray,
                   @ColumnInfo(name = "version")
                   val version: String) {
    override fun toString(): String {
        return filename
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Report

        if (id != other.id) return false
        if (filename != other.filename) return false
        if (!report.contentEquals(other.report)) return false
        if (version != other.version) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id
        result = 31 * result + filename.hashCode()
        result = 31 * result + report.contentHashCode()
        result = 31 * result + version.hashCode()
        return result
    }
}

File: Source/GUI/UWP/Report.cpp
/*  Copyright (c) MediaArea.net SARL. All Rights Reserved.
*
*  Use of this source code is governed by a BSD-style license that can
*  be found in the License.html file in the root of the source tree.
*/

//---------------------------------------------------------------------------
#include "pch.h"
#include "Report.h"

//---------------------------------------------------------------------------
using namespace MediaInfo;

using namespace Windows::Foundation;
using namespace Windows::Storage::Streams;
using namespace Windows::Security::Cryptography;

//---------------------------------------------------------------------------
// Constructor/Destructor
//---------------------------------------------------------------------------

//---------------------------------------------------------------------------
Report::Report(uint64 Id, Platform::String^ FileName, Platform::String^ Data, Platform::String^ Version) : _Id(Id), _FileName(FileName), _Data(Data), _Version(Version)
{
}

//---------------------------------------------------------------------------
Report::Report(Report^ Other)
{
    _Id=Other->Id;
    _FileName=ref new Platform::String(Other->FileName->Data());
    _Data=ref new Platform::String(Other->Data->Data());
    _Version=ref new Platform::String(Other->Version->Data());
}